src/OpenFeature/Providers/Memory/InMemoryProvider.cs

```diff
@@ -103,7 +102,7 @@ private ResolutionDetails<T> Resolve<T>(string flagKey, T defaultValue, Evaluati
         {
             if (!this._flags.TryGetValue(flagKey, out var flag))
             {
-                throw new FlagNotFoundException($"flag {flagKey} not found");
+                return new ResolutionDetails<T>(flagKey, defaultValue, ErrorType.FlagNotFound, Reason.Error);
```

Comment:
I changed this reference to not throw an exception but return `FlagNotFound`. Please let me know if we want to change this behaviour.

Reply:
Yes, it's better to avoid throwing is possible but please ensure the error hook is still run when the flag isn't found.

Reply:
I just rechecked the code execution. When the evaluation returns an Error, it will always call the hook. Check https://github.com/open-feature/dotnet-sdk/blob/c9e5ded5a36b3130b2743c75f408ffbfa5842edd/src/OpenFeature/OpenFeatureClient.cs#L279